source/auth/digest.ts

```diff
@@ -5,8 +5,12 @@ import { DigestContext } from "../types.js";
 const NONCE_CHARS = "abcdef0123456789";
 const NONCE_SIZE = 32;
 
-export function createDigestContext(username: string, password: string): DigestContext {
-    return { username, password, nc: 0, algorithm: "md5", hasDigestAuth: false };
+export function createDigestContext(
+    username: string,
+    password: string,
+    ha1: string
+): DigestContext {
+    return { username, password, ha1, nc: 0, algorithm: "md5", hasDigestAuth: false };
```

Comment:
Would you mind merging/rebasing master? I added formatting (prettier) tests.. want to make sure these changes still pass.